### PR TITLE
HDDS-7921. Migrate TestKeyManagerImpl to JUnit5

### PR DESCRIPTION
**Pull request by kaijchen**
## What changes were proposed in this pull request?

HDDS-7871 (#4236) needs to add a JUnit 5 test to TestKeyManagerImpl.

Better to do the migration first.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7921

## How was this patch tested?

Full CI: https://github.com/kaijchen/ozone/actions/runs/4120563700

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java [hunk 1]
```diff
@@ -638,8 +633,6 @@ public void testInvalidPrefixAcl() throws IOException {
         .build();
 
     // add acl with invalid prefix name
```
Comment (kaijchen) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java:635]: Hi @adoroszlai @hemantk-12 can you help review this PR?
Especially the following lines in `testInvalidPrefixAcl()`.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java [hunk 2]
```diff
@@ -638,8 +633,6 @@ public void testInvalidPrefixAcl() throws IOException {
...
-    exception.expect(OMException.class);
-    exception.expectMessage("Invalid prefix name");
     writeClient.addAcl(ozInvalidPrefix, ozAcl1);
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java:636]: Isn't this (and `setAcl`, `removeAcl`) expected to throw exception?  I see the test passes as is, but why?  Are these writes no-ops in the test?  If so, I think we should remove them.

Comment (kaijchen) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java:636]: It's calling the real method, but seems only `getAcl()` is validating the prefix.

https://github.com/apache/ozone/blob/11fd5b12fd73b0d47dcf53a91346d842658c56d7/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/PrefixManagerImpl.java#L99-L100

**Review by adoroszlai — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @kaijchen for the patch.

**Comment by kaijchen** [comment]
Thanks @adoroszlai for the review.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java (hunk 2 above)
Comment (kaijchen) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java:636]: Hi @adoroszlai, it turns out the exception should be thrown here.
There must be some errors on the master branch, see:
https://github.com/apache/ozone/pull/4396/files#r1135258057

Comment (kaijchen) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java:636]: > it turns out the exception should be thrown here.
> There must be some errors on the master branch, see:

HDDS-8157 is created to track this issue.

**Pull request merged**